src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/Licence.kt

```diff
@@ -187,4 +188,315 @@ data class Licence(
+    if (forename != other.forename) return false
+    if (middleNames != other.middleNames) return false
+    if (surname != other.surname) return false
+    if (dateOfBirth != other.dateOfBirth) return false
+    if (conditionalReleaseDate != other.conditionalReleaseDate) return false
+    if (actualReleaseDate != other.actualReleaseDate) return false
+    if (sentenceStartDate != other.sentenceStartDate) return false
+    if (sentenceEndDate != other.sentenceEndDate) return false
+    if (licenceStartDate != other.licenceStartDate) return false
+    if (licenceExpiryDate != other.licenceExpiryDate) return false
+    if (licenceActivatedDate != other.licenceActivatedDate) return false
+    if (topupSupervisionStartDate != other.topupSupervisionStartDate) return false
+    if (topupSupervisionExpiryDate != other.topupSupervisionExpiryDate) return false
+    if (probationAreaCode != other.probationAreaCode) return false
+    if (probationAreaDescription != other.probationAreaDescription) return false
+    if (probationPduCode != other.probationPduCode) return false
+    if (probationPduDescription != other.probationPduDescription) return false
+    if (probationLauCode != other.probationLauCode) return false
+    if (probationLauDescription != other.probationLauDescription) return false
+    if (probationTeamCode != other.probationTeamCode) return false
+    if (probationTeamDescription != other.probationTeamDescription) return false
+    if (appointmentPerson != other.appointmentPerson) return false
+    if (appointmentTime != other.appointmentTime) return false
+    if (appointmentAddress != other.appointmentAddress) return false
+    if (appointmentContact != other.appointmentContact) return false
+    if (spoDiscussion != other.spoDiscussion) return false
+    if (vloDiscussion != other.vloDiscussion) return false
+    if (approvedDate != other.approvedDate) return false
+    if (approvedByUsername != other.approvedByUsername) return false
+    if (approvedByName != other.approvedByName) return false
+    if (supersededDate != other.supersededDate) return false
+    if (submittedDate != other.submittedDate) return false
+    if (dateCreated != other.dateCreated) return false
+    if (dateLastUpdated != other.dateLastUpdated) return false
+    if (updatedByUsername != other.updatedByUsername) return false
+    if (standardConditions != other.standardConditions) return false
+    if (additionalConditions != other.additionalConditions) return false
+    if (bespokeConditions != other.bespokeConditions) return false
+    if (responsibleCom != other.responsibleCom) return false
+    if (submittedBy != other.submittedBy) return false
+    if (createdBy != other.createdBy) return false
+    if (variationOfId != other.variationOfId) return false
+    if (versionOfId != other.versionOfId) return false
+    if (licenceVersion != other.licenceVersion) return false
+
+    return true
+  }
+
+  override fun hashCode(): Int {
```

Comment:
this was generated against the java version of the class. The kotlin generator version was really complex.